### DISCUSSION TITLE
Version the default starter template, because now it is not a great experience for newcomers who try to follow the tutorial

### DISCUSSION THREAD

**Discussion by crosscode-nl** [gatsbyjs/gatsby#30383] [Help]
I'm trying to learn how to build websites with Gatsbyjs. It recently got updated to version 3, and not all plugins - like theme-ui - are working yet.

I learned that you can generate and use starter templates using the gatsby-cli tool. 

I expected that downgrading the gatsby-cli tool would use a starter template matching the version of the cli tool. It does not. It still generates a Gatsby 3.x site. 

Downgrading is hard because of two reasons: 

- Dependencies. I do not know which versions of dependencies I need to use. Maybe I need to use other dependencies for older versions of Gatsby. One way to find out would be to look at an older version of the start template.
- The default starter template is not versioned with semver tags in the repository. I would have to look at and understand each and every commit to get to a version I could use as a v2.x starter template. 

I’m not saying that the version 3.0 template does not work out of the box. It does. It just does not work well with every plugin yet, like theme-ui. 

The expected downgrade path - use an older version of the Gatsby cli - does not work. 

Gatsby-cli developers could have just used semver to tag the starter template and gatsby cli and use this version information to match the best starter template version to the version of cli. 

So, please please please, use semver https://semver.org/ to version your code in public repositories at least. 

Matching templates with the version of the cli and gatsby would give the best experience.

**Comment by LekoArts** [comment]
Hi, thanks for the comment!

Most plugins, including Theme UI, should continue to work with Gatsby v3 -- they only might print a peerDep warning. Theme UI itself is a special case as they are still on the 0.x version range which makes correct semver versioning difficult.

The Gatsby CLI clones public GitHub repositories (e.g. https://github.com/gatsbyjs/gatsby-starter-minimal) when setting up a new site so the versioning of the CLI isn't coupled with the versions of the starters. We want to give our users the latest versions available when setting up a new site so we always default to the new version.

You can find the v2 version of the starter here: https://github.com/gatsbyjs/gatsby-starter-minimal/tree/ca8ff417c12f1c6715b2abdafb371355ef6ae00f

The starters are not versioned with semver as they are not something people install but rather clone and then forget about it. For reusable Gatsby sites we have themes: https://www.gatsbyjs.com/docs/themes/ -- all our owned plugins and packages follow semver, so if you want to continue using Gatsby v2 you can use the last major version of every plugin we own.
